### PR TITLE
Keep the draft when opening the agents view

### PR DESCRIPTION
**Pull request by snimu**
## What this does

You can now open the agents view while you have an unsent draft in the prompt box. The draft is saved automatically and put back into the editor when you return to that session, with a short "Restored stashed prompt" notice.

Previously, both ways into the agents view (the subagents line under the prompt and the full view) refused with "Send, stash, or clear your draft before opening agents", forcing you to deal with the draft first.

## Why

The refusal existed to protect drafts from being lost when the chat view is swapped out. But the protection it asked you to do by hand — stashing — already exists as a mechanism the app manages itself. There was no reason to make it the user's problem.

## How it works

- When you head to the agents view with a draft, the draft (including text, images, and paste state) is saved into the same per-session stash that the manual stash feature uses, marked to be restored automatically.
- When you come back to that session, the draft is placed back into the editor if it is empty.
- If you had already stashed something manually, it is kept — the automatic one lines up in front of it instead of overwriting it.
- Pressing left inside a non-empty draft still moves the cursor; nothing about typing changed.

## Cleanup

The two refusal checks and their message are removed; the test asserting the refusal now asserts the new behavior. The manual stash action stays, since it is useful on its own.

## Changes

Lines changed: source +19/−9 (net +10 — the refusal checks went away, the restore-on-open path is new), tests +108/−3, changelog +1/−0.

## Checks

- `npm run check` clean; all touched test files pass, with new tests for the round-trip, manual-stash collision, and no-restore-when-not-flagged cases.
- Went through three review rounds; the first two caught real edge cases (whitespace-only drafts, one uncovered entrance, a double-save on rapid re-entry) that the final version handles.
- Validated end-to-end in a real terminal session: type a draft, open the subagents view, come back — the draft is intact; same for the full agents view and across both entrances.

---

> [!NOTE]
> **Low Risk**
> Localized interactive-mode prompt stash behavior with no auth, persistence, or network changes; risk is mainly UX edge cases around stash ordering and status display.
> 
> **Overview**
> Opening the **full or scoped agents view** no longer blocks when the prompt has an unsent draft. The draft is **auto-stashed** (text, images, paste state) with a `restoreOnOpen` flag and put back in the editor when you return to that session.
> 
> `returnToAgentsView` calls `stashDraftForAgentsView` before teardown; `run()` calls `restorePromptStashOnOpen` after `init()` so only auto-stashes restore on reopen—manual stashes stay manual. An existing manual stash is **queued behind** the auto-stash instead of being overwritten. Whitespace-only drafts are not auto-stashed.
> 
> The old **"Send, stash, or clear your draft before opening agents"** checks on `requestAgentsView` and `openScopedAgentsView` are removed.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f591d243c2bca223c14fdfd9371ee410a1b5bd16. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Keep draft prompt when navigating to the agents view by auto-stashing and restoring it
> - Previously, opening the agents view (full or scoped) was blocked when the editor contained a draft; now the draft is auto-stashed with `restoreOnOpen: true` and restored when the session reopens.
> - Adds `stashDraftForAgentsView()` in [interactive-mode.ts](https://github.com/PrimeIntellect-ai/prime-agent/pull/1372/files#diff-b96137f89d422953665b334785273ae6cdbcb53ff37d8828db43f85038658316) to snapshot the draft before teardown; any existing manual stash is queued behind it.
> - Adds `restorePromptStashOnOpen()`, called after `init()`, to restore the stash into an empty editor using a fresh status block.
> - Whitespace-only drafts are ignored and manual stashes are never auto-restored on open.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized f591d24.</sup>
> 
>

### CODE REVIEW

File: packages/coding-agent/src/modes/interactive/interactive-mode.ts [hunk 1]
```diff
@@ -1464,6 +1464,7 @@ export class InteractiveMode {
 
 	async run(): Promise<InteractiveModeRunResult> {
 		await this.init();
+		this.restorePromptStashOnOpen();
```
[the remark this answers is not in the document]
Comment (snimu) [reply packages/coding-agent/src/modes/interactive/interactive-mode.ts:1467]: Real — showStatus replaces the anchored last status, so the restore notice could overwrite a notice init just posted (e.g. compaction). Fixed in 6ae202823: the on-open restore drops the status anchor first, so its notice lands in a fresh block and both messages stay visible. The other restore paths keep the replace behavior — only the on-open path races init's statuses.

**Review by sethkarten — APPROVED** [review]
